### PR TITLE
Role issue

### PR DESCRIPTION
**Pull request by Gusted**
- Avoid re-assignments `xxx = xxx.WithRole(xxx)`.
- Enable `modifies-value-receiver` rule(this was the only error)

### CODE REVIEW

**Comment by wxiaoguang** [comment]
In Golang, `c2 = WithContext(c1)` means `c1` is unchanged and returns a new `c2`.

So `WithRole` seems not following the naming style here. Perhaps we can think a new name for it.

**Comment by lunny** [comment]
> * Avoid re-assignments `xxx = xxx.WithRole(xxx)`.
> * Enable `modifies-value-receiver` rule(this was the only error)

I against this. `Int` is efficient enough. We don't need the pointer. No benefit from the refactor.

File: models/issue_comment.go [hunk 1]
```diff
@@ -122,9 +122,8 @@ const (
 )
 
 // WithRole enable a specific tag on the RoleDescriptor.
-func (rd RoleDescriptor) WithRole(role RoleDescriptor) RoleDescriptor {
-	rd |= (1 << role)
```
Comment (lunny) [inline models/issue_comment.go:126]: I think we need to do is change this line to `return rd | (1 << role)` and remove the next line.

**Comment by wxiaoguang** [comment]
Or could we just disable the lint for RoleDescriptor? The old code seems fine. :)

**Comment by Gusted** [comment]
> The old code seems fine. :)

The old code is fine indeed, after applying the suggestion from lunny(which is the actual correct fix for the linter).

**Review by lunny — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**